### ISSUE TITLE
Video stream became 'broken' randomly

### ISSUE THREAD

**Issue by bymet** [aiortc/aiortc#182]
Streaming can work fine for 10 seconds or 10 minutes, but then suddenly it becomes like that:
![image](https://user-images.githubusercontent.com/33255173/59494902-da498280-8e96-11e9-931b-fc87b55356cd.png)

Eventually it can fix itself, but sometime it does not. I have tried to stream from android device and browser and got same behavior. Any idea what can be wrong?

Frame processing looks like that:
```
async def recv(self):
        frame = await self.track.recv()

        if (self.transform == "nothing"):
            return frame

        try:
            networkWidth = self.detectionService.width
            networkHeight = self.detectionService.height
            
            widthScale = frame.width / networkWidth
            heightScale = frame.height / networkHeight

            frame_rgb = cv2.cvtColor(frame.to_ndarray(format="bgr24"), cv2.COLOR_BGR2RGB)
            frame_resized = cv2.resize(frame_rgb, (networkWidth, networkHeight), interpolation=cv2.INTER_LINEAR)

            darknet.copy_image_from_bytes(self.darknet_image, frame_resized.tobytes())
            self.lastDetection = self.detectionService.detect(self.darknet_image)
            self.detections.put(self.lastDetection)

            image = cvDrawBoxes(self.lastDetection, frame_rgb, widthScale, heightScale)
            new_frame = VideoFrame.from_ndarray(image, format="rgb24")
            new_frame.pts = frame.pts
            new_frame.time_base = frame.time_base

            return new_frame
        except:
            info = sys.exc_info()
            return frame;            
```

**Comment by jlaine** [comment]
What is the screenshot I'm looking at? Something received by aiortc? Sent by aiortc?

The corruption you are showing looks like packet loss, so you'll want to figure out why you are losing packets.

Concerning the code pasted above, I'm not convinced it will run it realtime unless you use a low framerate + video size and / or offload the image processing to a separate thread or process. If you are doing 30fps you have at most 30ms to process each image, is this the case?

**Comment by jlaine** [comment]
Hello?

**Comment by andriy-tsvikla** [comment]
I am facing similar issue.   stream 320x240 4 fps

**Comment by jlaine** [comment]
I'm going to close this issue for lack of response from the original submitter - I have nothing to go on here.

**Comment by jlaine** [comment]
Should be fixed by 42f4e0f57b186aa15b0e731be37ca35afd780100
